modules/videoio/src/cap_android_mediandk.cpp

```diff
@@ -154,30 +153,58 @@ class AndroidMediaNdkCapture : public IVideoCapture
         if (buffer.empty()) {
             return false;
         }
-        Mat yuv(frameHeight + frameHeight/2, frameWidth, CV_8UC1, buffer.data());
+
+        Mat yuv(frameHeight + frameHeight/2, frameStride, CV_8UC1, buffer.data());
+
         if (colorFormat == COLOR_FormatYUV420Planar) {
-            cv::cvtColor(yuv, out, cv::COLOR_YUV2BGR_YV12);
+            cv::cvtColor(yuv, frame, cv::COLOR_YUV2BGR_YV12);
         } else if (colorFormat == COLOR_FormatYUV420SemiPlanar) {
-            cv::cvtColor(yuv, out, cv::COLOR_YUV2BGR_NV21);
+            cv::cvtColor(yuv, frame, cv::COLOR_YUV2BGR_NV21);
         } else {
             LOGE("Unsupported video format: %d", colorFormat);
             return false;
         }
+
+        Mat croppedFrame = frame(Rect(0, 0, videoWidth, videoHeight));
+        out.assign(croppedFrame);
```

Comment:
This is not needed with proper constructor.

Reply:
Actually for a video 1920 x 1080 the frame size is actually 1920 x 1088 (the height and width are aligned to a multiple of 16). The proposed construction should solve the problem with aligned width but not the aligned height.
Is there another way to "cut" rows at the end of a Mat ?

Reply:
Got it, thanks for the clarification!